src/transport_performance/population/rasterpop.py

```diff
@@ -0,0 +1,707 @@
+            hexstring.
+        boundary_weight : float, optional
+            Weight (in pixels) of the boudary lines, by default 2.
+
+        Returns
+        -------
+        folium.Map
+            Folium map obeject, with data and boundaries in layers. Will be
+            None when writing to file (saves time when displaying map
+            interactive mode).
+
+        Notes
+        -----
+        1. When using tile layers, check the tile provider's terms and
+        conditions and assign an attribution as needed.
+
+        References
+        ----------
+        .. [1] https://python-visualization.github.io/folium/modules.html
+        .. [2] http://leaflet-extras.github.io/leaflet-providers/preview/
+        .. [3] https://matplotlib.org/stable/tutorials/colors/colormaps.html
+
+        """
+        # add a base map with no tile - then is it not on a layer control
+        m = folium.Map(tiles=None, control_scale=True, zoom_control=True)
+
+        # add a tile layer
+        folium.TileLayer(
+            tiles=tiles,
+            attr=attr,
+            show=False,
+            control=False,
+        ).add_to(m)
+
+        # add the variable data to the map
+        self.pop_gdf.explore(
+            self.__var_name, cmap=cmap, name=self.__var_name.capitalize(), m=m
+        )
+
+        # add the urban centre boundary, if one was provided
+        if self._uc_gdf is not None:
+            self._uc_gdf.explore(
+                tooltip=["boundary"],
+                name="Urban Centre Boundary",
+                m=m,
+                style_kwds={
+                    "fill": False,
+                    "color": boundary_color,
+                    "weight": boundary_weight,
+                },
+            )
+
+        # add the area of interest boundary
+        self._aoi_gdf.explore(
+            tooltip=["boundary"],
+            name="Area of Interest Boundary",
+            m=m,
+            style_kwds={
+                "fill": False,
+                "color": boundary_color,
+                "weight": boundary_weight,
+                "dashArray": 7,
+            },
+        )
+
+        # add the centroids to a separate layer
+        self.centroid_gdf.explore(
+            self.__UC_COL_NAME,
+            name="Centroids",
+            m=m,
+            show=False,
+            style_kwds={
+                "style_function": lambda x: {
+                    "color": "#BC544B"
+                    if x["properties"][self.__UC_COL_NAME] is False
+                    else "#8B0000"
+                }
+            },
+            legend=False,
+        )
+
+        # fit bounds to the plot limits and add a layer control button
+        m.fit_bounds(m.get_bounds())
+        m.add_child(folium.LayerControl())
+
+        # write to file if filepath is given
+        if save is not None:
+            out_df = os.path.dirname(save)
+            if not os.path.exists(out_df):
+                os.mkdir(out_df)
```

Comment:
Applies to all backend types in `RasterPop.plot()`:

Logic breaks when user tries:
```
>>> self.plot(which="cartopy", save="foobar.png")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 150, in plot
  File "<stdin>", line 560, in _plot_cartopy
```
Os.mkdir doesn’t like cases where the parent directory returned equals “”.  [There is a utility that should do what you need](https://github.com/datasciencecampus/transport-network-performance/blob/9cb601de0461002f71c39e6a79830db70d4b4131/src/transport_performance/utils/defence.py#L31C1-L32C1), yet to be merged to dev though.

Example implementation:

```
from transport_performance.utils.defence import _check_parent_dir_exists

…

_check_parent_dir_exists(save, ”save”, create=True)
```